### DISCUSSION TITLE
Create New File Version

### DISCUSSION THREAD

**Discussion by gnevnev** [koechlm/Vault-Job-Processor---iLogic-Extension#31] [Q&A] [answered]
Hi,

I'm using this extension with a master rule to run 1 of two rules depending on whether it is a drawing or part/assembly. On drawings it saves the partslist table to an excel file, on parts/assemblies it moves a pre-existing rendered image of the part to another location.
Occasionally the job processor will create a new file version even though the 'Create New File Version' box is unticked on all rules in iLogic administration. Unfortunately when it does create a new file version this prevents other job processor tasks from running afterwards 'Sync properties not allowed on non-tip versions'.
This happens to both drawing and part files but I'm having a hard time narrowing down the cause.
'Created by Custom Job executing iLogic: External Rule: VaultJobRule_Master.iLogicVb' is the comment on the affected parts/drawings.

I have attached the master and two sub rules. Any suggestions on what could be the cause of this would be greatly appreciated.

[VaultJobRules.zip](https://github.com/koechlm/Vault-Job-Processor---iLogic-Extension/files/13241370/VaultJobRules.zip)

Thanks,
George

**Comment by koechlm** [comment, marked as answer]
Hi George,
User rules and life cycle rules behave differently:

  *   User rules allow to administratively set the option “check in new file version”, whereas life cycle rules always check-out the file
  *   Review comments in code lines #76 and #713
Your rules might not change the files; Vault recognizes changes comparing checksum + modification date. If the job saved the file, but did not really change something (checksum) the file will return without version increment.
This explains, why you see some jobs iterating the file, others not.

Your problem – subsequent Property Update jobs fail – is a known limitation and I already have put a change on the future enhancement lists: the iLogic-Job Extension should get an option to submit the Property Update if a file changed; this will result in submitting the latest revision (tip version) for Property Update.

I hope this helps.
Best regards,
Markus

**Reply by gnevnev** [reply]
Hi Markus,

Thankyou so much for the reply. This really clears things up and I'm glad to see that this behaviour is all as expected.
Glad to hear that this is already a future enhancement, when working on large assemblies the number of failed jobs can be quite intimidating.

That being said this addin has significantly improved our workflow so thankyou for taking the time to create it!

Thanks,
George

**Reply by gnevnev** [reply]
Hi Markus,

Out of interest is there a reason behind the lifecycle rule not having the same 'create new file version' option as the user added rules?
Is this something that could be added for a future release?
In our case we do not need any changes to the file to be saved, but appreciate that this is not the case for everyone .

Thanks,
George :)

**Reply by koechlm** [reply]
Hi George,
I don't see any preventing reason so that I will consider this for future release(s). See https://github.com/koechlm/Vault-Job-Processor---iLogic-Extension/issues/32.
Thank you for your appreciated feedback!
Markus
